index.js

```diff
@@ -43,6 +43,11 @@ exports.connect = function (URL, options, callback) {
 		URL.extraHeaders = options.extraHeaders
 	}
 
+	if (options.hasOwnProperty('protocols')){
+		URL.extraHeaders = URL.extraHeaders || {}
+		URL.extraHeaders['Sec-WebSocket-Protocol'] = options.protocols.join(', ')
+	}
+
```

Comment:
From spec:
> WebSocket clients
   verify that the server included one of the values that was specified
   in the WebSocket client's handshake.

Our client should check that the server pick one of the sub-protocols provided by the user.